backend/scheduler/tasks.py

```diff
@@ -57,13 +59,24 @@ def execute_pipeline_task(
     name: Any,
 ) -> None:
     logger.info(f"Executing pipeline name: {name}")
-
     try:
         logger.info(f"Executing workflow id: {workflow_id}")
         tenant: Organization = (
             get_tenant_model().objects.filter(schema_name=org_schema).first()
         )
         with tenant_context(tenant):
+            subscription_plugin_available = (
+                os.environ.get("SUBSCRIPTION_PLUGIN_AVAILABLE", "False") == "True"
+            )
+            logger.info(f"Is subscription available: {subscription_plugin_available}")
+
+            if subscription_plugin_available and not etl_prerun_check(org_schema):
```

Comment:
@athul-rs you anyway seem to check for this env within `etl_prerun_check(org_schema)`. You can move all of this over there and simplify code here

Reply:
This check needs to be here in celery task, removed from the other method since it is already checked here.